src/panels/lovelace/cards/hui-markdown-card.ts

```diff
@@ -111,7 +111,7 @@ export class HuiMarkdownCard extends LitElement implements LovelaceCard {
       this._config.show_empty === false &&
       this._templateResult.result.length === 0;
     if (shouldBeHidden !== this.hidden) {
-      this.style.display = shouldBeHidden ? "none" : "block";
+      this.style.display = shouldBeHidden ? "none" : "";
       this.toggleAttribute("hidden", shouldBeHidden);
```

Comment:
Removing the line setting the `display` property would IMO work too, as the HTML `hidden` attribute is toggled as well. Is there a reason to do both?

Reply:
hidden is use by the parent (grid card) to know if it's hidden or not so it can ignore it. The display css property is used to not display the element. (Semantic vs styling, we need both)